api/submit_result.go

```diff
@@ -1,12 +1,16 @@
 package api
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
 	"github.com/mesg-foundation/core/pubsub"
 	"github.com/mesg-foundation/core/service"
 )
 
 // SubmitResult submits results for executionID.
-func (a *API) SubmitResult(executionID string, outputKey string, outputData map[string]interface{}) error {
+func (a *API) SubmitResult(executionID string, outputKey string, outputData interface{}) error {
```

Comment:
We talked in the meeting to use `[]byte` here. I could be quite ok with the `interface{}` which was my initial idea too. I'm curious to know why you put that ? Is their any places we still send a `map[string]interface{}`, I only see the grpc api. If there is only this I think we shouldn't overcomplicated and just keep one kind of data instead of having to manage string and map[string]interface{}.

But here again i'm more curious about the reason of that.

Reply:
for the WSS we faced with a similar problem where which explained here https://github.com/mesg-foundation/core/issues/564. But we don't need to accept here in the _service_ api as you said because we don't make calls to inside Core, only service clients use it. So, I'm changing it to bytes and we can always switch to interface if needed.